tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -240,7 +252,94 @@ export const createNewFileOrFolder = async (args: createResourceArgs): Promise<v
       await page.goto(page.url())
       break
     }
+    case 'OpenDocument': {
+      // By Default when OpenDocument is created, it is opened with collabora if both app-provider services are running together
+      await createOpenDocumentOrMicrosoftWordFile(args, 'Collabora')
+      break
+    }
+    case 'Microsoft Word': {
+      // By Default when Microsoft Word document is created, it is opened with OnlyOffice if both app-provider services are running together
+      await createOpenDocumentOrMicrosoftWordFile(args, 'OnlyOffice')
+      break
+    }
+  }
+}
+
+const createOpenDocumentOrMicrosoftWordFile = async (
+  args: createResourceArgs,
+  editorToOpen: string
+): Promise<void> => {
+  const { page, name, type, content } = args
+  await page.locator(util.format(createNewOpenDocumentFileButton, type)).click()
+  await page.locator(resourceNameInput).fill(name)
+  let newEditorTab
+  await Promise.all([
+    (newEditorTab = page.waitForEvent('popup')),
+    page.waitForResponse((resp) => resp.status() === 200 && resp.request().method() === 'POST'),
+    page.locator(util.format(actionConfirmationButton, 'Create')).click()
+  ])
+  const editorPage = await newEditorTab
+  await editorPage.waitForLoadState()
+  await editorPage.waitForURL('**/external/personal/**')
+  const editorMainFrame = await editorPage.frameLocator(editorIframe)
+  switch (editorToOpen) {
+    case 'Collabora':
+      await editorMainFrame.locator(welcomeModalCollaboraIframe).waitFor()
+      await editorPage.keyboard.press('Escape')
+      await editorMainFrame.locator(openDocumentTextAreaSelector).type(content)
+      const saveModified = await editorMainFrame.locator(savedInCollaboraEditorSelector)
+      await expect(saveModified).toBeVisible()
+      await editorMainFrame.locator('#Save').click()
+      await expect(saveModified).not.toBeVisible()
+      break
+    case 'OnlyOffice':
+      const innerIframe = await editorMainFrame.frameLocator(onlyOfficeInnerFrameSelector)
+      await innerIframe.locator(microsoftWordTextAreaSelector).type(content)
+      const saveButtonDisabledLocator = innerIframe.locator(microsoftWordSaveButtonSelector)
+      await expect(saveButtonDisabledLocator).toHaveAttribute('disabled', 'disabled')
+      break
+    default:
+      throw new Error(
+        "Editor should be either 'Collabora' or 'OnlyOffice' but found " + editorToOpen
+      )
+  }
+  await editorPage.close()
+}
+
+export const openAndGetContentOfOpenDocumentOrMicrosoftWordDocument = async ({
+  page,
+  editorToOpen
+}: {
+  page: Page
+  editorToOpen: string
+}): Promise<string> => {
+  let newEditorTab
+  newEditorTab = page.waitForEvent('popup')
+  const editorPage = await newEditorTab
```

Comment:
Maybe wait for `popup` with some action?
```suggestion
  const editorPage = await page.waitForEvent('popup')
```